lib/benchmarks/scroll.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+const Duration _animationCheckingInterval = Duration(milliseconds: 50);
+
+const Duration _scrollAnimationLength = Duration(milliseconds: 300);
+
+Offset _absoluteTopLeft(RenderObject renderObject) =>
+    (renderObject as RenderBox).localToGlobal(Offset.zero);
+
+Rect _absoluteRect(RenderObject renderObject) =>
+    _absoluteTopLeft(renderObject) & renderObject.paintBounds.size;
+
+Size _windowSize(BuildContext context) => MediaQuery.of(context).size;
+
+Rect _windowRect(BuildContext context) => Offset.zero & _windowSize(context);
+
+bool _isSuperset({@required Rect large, @required Rect small}) =>
```

Comment:
This could also be `large.intersect(small) == small`, but maybe what you have here is more clear

Reply:
I agree that `large.intersect(small) == small` is more concise, but I don't know if it would cause any floating-point precision issues.